cassandra-leaves-app/src/pages/index.js

```diff
@@ -58,85 +72,125 @@ export default function Home({ leavesData }) {
 
-            {/* Table */}
-            <table>
-                <thead>
-                <tr>
-                    <th>ID</th>
-                    <th>Title</th>
-                    <th>Domain</th>
-                    <th>Actions</th>
-                </tr>
-                </thead>
-                <tbody>
+            {isModalOpen && (
+                <div className="modal">
+                    <div className="modal-content">
+                        <h2>{isEditing ? "Edit Record" : "Add New Record"}</h2>
+                        <input
+                            type="text"
+                            name="title"
+                            placeholder="Title"
+                            value={formState.title}
+                            onChange={handleInputChange}
+                        />
+                        <input
+                            type="text"
+                            name="domain_name"
+                            placeholder="Domain"
+                            value={formState.domain_name}
+                            onChange={handleInputChange}
+                        />
+                        <div className="modal-actions">
+                            {isEditing ? (
+                                <button onClick={handleUpdateRecord}>Update</button>
+                            ) : (
+                                <button onClick={handleAddRecord}>Add</button>
+                            )}
+                            <button onClick={() => setIsModalOpen(false)}>Cancel</button>
+                        </div>
+                    </div>
+                </div>
+            )}
+
+            {isDeleteModalOpen && (
+                <div className="modal">
+                    <div className="modal-content">
+                        <h2>Confirm Deletion</h2>
+                        <p>Are you sure you want to delete this record?</p>
+                        <div className="modal-actions">
+                            <button onClick={confirmDelete}>Yes, Delete</button>
+                            <button onClick={cancelDelete}>Cancel</button>
+                        </div>
+                    </div>
+                </div>
+            )}
+
+            <div className="grid">
                 {paginatedData.map((item) => (
-                    <tr key={item.id}>
-                        <td>{item.id}</td>
-                        <td dangerouslySetInnerHTML={{ __html: item.title }}></td>
-                        <td>{item.domain_name}</td>
-                        <td>
-                            <button onClick={() => handleEdit(item)}>Edit</button>
-                            <button onClick={() => handleDelete(item.id)}>Delete</button>
-                            <Link href={`/quote/${item.id}`}>
-                                <button>View</button>
-                            </Link>
-                        </td>
-                    </tr>
+                    <Card
```

Comment:
Great job on creating a well-structured and functional UI! The return statement is clear, and the code is readable. Below are some suggestions to improve the organization, scalability, and overall quality of this part of the code:

### 1. **Break Down the Return Statement**
The return statement is handling a lot of logic directly, which can make it harder to maintain as the component grows. Consider breaking it into smaller, reusable sub-components. For example:
- **Modal Components**: Create separate `Modal` and `DeleteModal` components to handle the modal logic and UI.
- **Pagination Component**: Extract the pagination controls into a `Pagination` component.

Example:
```tsx
<Pagination
    currentPage={currentPage}
    totalPages={totalPages}
    onPrevious={() => setCurrentPage((prev) => Math.max(prev - 1, 1))}
    onNext={() => setCurrentPage((prev) => Math.min(prev + 1, totalPages))}
/>
```

---

### 2. **Abstract Logic into Custom Hooks**
Some logic (e.g., modals, pagination) could be abstracted into custom hooks to keep the return statement cleaner:
- **Use `useModal` for modal state and handlers.**
- **Use `usePagination` for pagination logic.**

This separation simplifies the JSX and makes the component easier to test and reuse.

---

### 3. **Conditional Rendering**
For better readability, extract conditional rendering blocks like modals into helper functions or separate components. For example:
```tsx
const renderModal = () => isModalOpen && (
    <Modal
        title={isEditing ? "Edit Record" : "Add New Record"}
        onClose={() => setIsModalOpen(false)}
        actions={
            isEditing
                ? <button onClick={handleUpdateRecord}>Update</button>
                : <button onClick={handleAddRecord}>Add</button>
        }
    >
        <input
            type="text"
            name="title"
            placeholder="Title"
            value={formState.title}
            onChange={handleInputChange}
        />
        <input
            type="text"
            name="domain_name"
            placeholder="Domain"
            value={formState.domain_name}
            onChange={handleInputChange}
        />
    </Modal>
);
```

---

### 4. **Move Static Content to Configurations**
The strings like `"Cassandra Leaves Dashboard"`, `"Add Record"`, and `"Confirm Deletion"` could be moved into a constants file or localization framework for easier maintenance and scalability.

---

### 5. **Centralize Dark Mode Logic**
The `isDarkMode` conditional class toggling (`className={`container ${isDarkMode ? "dark" : ""}`}`) can be centralized using a utility function:
```tsx
function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
}
```
Example:
```tsx
<div className={classNames("container", isDarkMode && "dark")}>
```

---

### 6. **Accessibility Enhancements**
- Ensure that interactive elements like the "Go to top" button and pagination controls are accessible:
    - Add `aria-label` or `role` attributes.
    - Use `<button>` for "Go to top" instead of `<div>`.
- Add focus management for modals (e.g., focus the first input when the modal opens).

---

### 7. **Styling Improvements**
Consider using a utility-first CSS framework like **Tailwind CSS** for styling. It will help simplify and standardize your styles, reducing the need for extensive CSS-in-JSX.

---

### 8. **Add Error Handling**
Currently, there’s no error boundary or fallback for potential issues (e.g., empty `leavesData`). Adding a fallback UI (like a loading spinner or error message) would improve the user experience.

---

### 9. **Optimize Images**
Ensure all images (e.g., `/up-chevron_8213555.png`) have proper dimensions and are optimized for performance. Use `next/image` if you're using Next.js to take advantage of image optimization.

---

These changes will make the component more modular, readable, and scalable, while also improving user experience and accessibility. Let me know if you'd like help with any specific part of the implementation!

Great work so far—this is shaping up to be a solid component! 🚀